util/clusterauth/clusterauth_test.go

```diff
@@ -86,25 +86,16 @@ func TestCreateServiceAccount(t *testing.T) {
 	})
 
 	t.Run("SA exists already", func(t *testing.T) {
-		cs := fake.NewSimpleClientset(ns, sa)
+		cs := fake.NewClientset(ns, sa)
 		err := CreateServiceAccount(cs, "argocd-manager", "kube-system")
 		require.NoError(t, err)
 		rsa, err := cs.CoreV1().ServiceAccounts("kube-system").Get(context.Background(), "argocd-manager", metav1.GetOptions{})
 		require.NoError(t, err)
 		assert.NotNil(t, rsa)
 	})
 
-	t.Run("Invalid name", func(t *testing.T) {
-		cs := fake.NewSimpleClientset(ns)
```

Comment:
The k8s test client no longer checks for invalid resource names, so this test fails.

Outside of tests, we always hardcode a valid resource name when we call `CreateServiceAccount`. This test is safe to remove.